analytics_data_api/utils.py

```diff
@@ -84,3 +94,92 @@ def date_range(start_date, end_date, delta=datetime.timedelta(days=1)):
     while cur_date < end_date:
         yield cur_date
         cur_date += delta
+
+
+def get_course_report_download_details(course_id, report_name):
+    """
+    Determine the path that the report file should be located at,
+    then return metadata sufficient for downloading it.
+    """
+    report_location_template = getattr(settings, 'COURSE_REPORT_FILE_LOCATION_TEMPLATE')
```

Comment:
To reduce the dependencies between this PR and your configuration PR (and to deal with the case where users simply update the analytics_api code without re-installing via configuration), we need to make sure that this code works even with the original analytics_api.yml file installed.

So this check needs to provide a default value, e.g.,

`report_location_template = getattr(settings, 'COURSE_REPORT_FILE_LOCATION_TEMPLATE', 'invalid-report-template')`